### PR TITLE
Add jwt verify

### PR DESCRIPTION
**Pull request by AlexHsieh**
- reference js jwt verify function
- all tests pass

### CODE REVIEW

File: Tests/Web5Tests/Crypto/JOSE/JWTTests.swift [hunk 1]
```diff
@@ -21,6 +22,133 @@ final class JWTTests: XCTestCase {
...
+            XCTAssertEqual(error, JWT.Error.verificationFailed("Expected JWT header to contain kid"))
+        }
+
+        // decode payload fails
+        let header = JWS.Header(algorithm: .es256k, keyID: "keyID", type: "JWT")
+        let base64UrlEncodedHeader = try JSONEncoder().encode(header).base64UrlEncodedString()
+        XCTAssertThrowsError(try JWT.parse(jwtString: "\(base64UrlEncodedHeader).efgh.hijk"))
+    }
+
+    func testVerifyJwtExpired() async throws {
```
Comment (kirahsapong) [inline Tests/Web5Tests/Crypto/JOSE/JWTTests.swift:55]:
```suggestion
    func test_verifyJwtExpired() async throws {
```

File: Tests/Web5Tests/Crypto/JOSE/JWTTests.swift [hunk 2]
```diff
@@ -21,6 +22,133 @@ final class JWTTests: XCTestCase {
...
+        let claims = JWT.Claims(expiration: Date(timeIntervalSinceNow: -1))
+        let base64UrlEncodedClaims = try JSONEncoder().encode(claims).base64UrlEncodedString()
+        do {
+             _ = try await JWT.verify(jwt: "\(base64UrlEncodedHeader).\(base64UrlEncodedClaims).hijk")
+        } catch let error as JWT.Error {
+            XCTAssertEqual(error, JWT.Error.verificationFailed("JWT has expired"))
+        }
+    }
+
+    func testVerifyJwtKeyIdNotDereferenceVerificationMethod() async throws {
```
Comment (kirahsapong) [inline Tests/Web5Tests/Crypto/JOSE/JWTTests.swift:68]:
```suggestion
    func test_verifyJwtKeyIdNotDereferencedVerificationMethod() async throws {
```

File: Tests/Web5Tests/Crypto/JOSE/JWTTests.swift [hunk 3]
```diff
@@ -21,6 +22,133 @@ final class JWTTests: XCTestCase {
...
+        let claims = JWT.Claims(issuer: did.uri, subject: did.uri, issuedAt: Date())
+        let base64UrlEncodedClaims = try JSONEncoder().encode(claims).base64UrlEncodedString()
+        do {
+             _ = try await JWT.verify(jwt: "\(base64UrlEncodedHeader).\(base64UrlEncodedClaims).hijk")
+        } catch let error as JWT.Error {
+            XCTAssertEqual(error, JWT.Error.verificationFailed("Expected kid in JWT header to dereference a DID Document Verification Method"))
+        }
+    }
+
+    func testVerifyPublicKeyAlgorithmIsNotSupported() async throws {
```
Comment (kirahsapong) [inline Tests/Web5Tests/Crypto/JOSE/JWTTests.swift:81]:
```suggestion
    func test_verifyPublicKeyAlgorithmIsNotSupported() async throws {
```

File: Tests/Web5Tests/Crypto/JOSE/JWTTests.swift [hunk 4]
```diff
@@ -21,6 +22,133 @@ final class JWTTests: XCTestCase {
...
+        let claims = JWT.Claims(issuer: did.uri, subject: did.uri, issuedAt: Date())
+        let base64UrlEncodedClaims = try JSONEncoder().encode(claims).base64UrlEncodedString()
+        do {
+             _ = try await JWT.verify(jwt: "\(base64UrlEncodedHeader).\(base64UrlEncodedClaims).hijk")
+        } catch let error as JWT.Error {
+            XCTAssertEqual(error, JWT.Error.verificationFailed("Expected alg in JWT header to match DID Document Verification Method alg"))
+        }
+    }
+
+    func testReturnSignerDidIfVerificationSucceeds() async throws {
```
Comment (kirahsapong) [inline Tests/Web5Tests/Crypto/JOSE/JWTTests.swift:94]:
```suggestion
    func test_verifyReturnDecodedJwt() async throws {
```

File: Tests/Web5Tests/Crypto/JOSE/JWTTests.swift [hunk 5]
```diff
@@ -21,6 +22,133 @@ final class JWTTests: XCTestCase {
...
+        let didStr = """
```
Comment (kirahsapong) [inline Tests/Web5Tests/Crypto/JOSE/JWTTests.swift:95]: We probably don't need to import a portable DID here. Could we `let did = try DIDJWK.create()` to ensure we are only testing the success of `verify` here?

**Review by kirahsapong — COMMENTED** [review]
Great work, and happy WWDC day! A couple of small comments, and then LGTM.

File: Tests/Web5Tests/Crypto/JOSE/JWTTests.swift (hunk 5 above)
Comment (AlexHsieh) [reply Tests/Web5Tests/Crypto/JOSE/JWTTests.swift:95]: Yes you are right. we don't need to import a portable DID here.

**Comment by AlexHsieh** [comment]
> Great work, and happy WWDC day! A couple of small comments, and then LGTM.

Happy WWDC. In the video of The platform state of the union shows xcode create a list view just use a line of text.. looks so handy. Also ah what is the abbreviation of LGTM?

**Comment by kirahsapong** [comment]
@AlexHsieh Yes! So many exciting new updates. 

Sorry - LGTM = looks good to me 😄 

Thanks for making that update. Great work 🎉

**Review by kirahsapong — APPROVED** [review]

**Pull request merged**